mantis-control-plane/mantis-control-plane-core/src/main/java/io/mantisrx/server/core/master/ZookeeperMasterMonitor.java

```diff
@@ -36,36 +37,45 @@
 /**
  * A monitor that monitors the status of Mantis masters.
  */
-public class ZookeeperMasterMonitor extends AbstractIdleService implements MasterMonitor {
+public class ZookeeperMasterMonitor extends BaseService implements MasterMonitor {
 
     private static final Logger logger = LoggerFactory.getLogger(ZookeeperMasterMonitor.class);
 
-    private final CuratorFramework curator;
+    private final CuratorService curator;
     private final String masterPath;
     private final BehaviorSubject<MasterDescription> masterSubject;
     private final AtomicReference<MasterDescription> latestMaster = new AtomicReference<>();
     private final NodeCache nodeMonitor;
-    private final JsonSerializer jsonSerializer;
 
-    public ZookeeperMasterMonitor(CuratorFramework curator, String masterPath) {
+    public ZookeeperMasterMonitor(CuratorService curator, String masterPath) {
         this.curator = curator;
         this.masterPath = masterPath;
         this.masterSubject = BehaviorSubject.create();
-        this.nodeMonitor = new NodeCache(curator, masterPath);
-        this.jsonSerializer = new JsonSerializer();
+        this.nodeMonitor = new NodeCache(curator.getCurator(), masterPath);
     }
 
     @Override
-    public void startUp() throws Exception {
+    public void start()  {
+        try {
+            if(curator.getCurator().getState() != CuratorFrameworkState.STARTED) {
```

Comment:
just checking: is this thread safe?

Reply:
ah, yeah, looks like it is not.  it does seem that there is the potential for racing "starts" to result in a thrown exception here.

i think a small refactor can fix this.  i'll do that today.

Reply:
oh wait, i take that back, concurrent calls to "start()" should be safe.  subsequent calls should be no-ops.